web/src/core/adapters/s3Client/s3Client.ts

```diff
@@ -606,20 +606,25 @@ export function createS3Client(
 
             const { awsS3Client } = await getAwsS3Client();
 
-            await awsS3Client.send(
-                new (await import("@aws-sdk/client-s3")).DeleteObjectsCommand({
-                    Bucket: bucketName,
-                    Delete: {
-                        Objects: paths.map(path => {
-                            const { objectName } =
-                                bucketNameAndObjectNameFromS3Path(path);
-                            return {
-                                Key: objectName
-                            };
-                        })
-                    }
-                })
-            );
+            try {
+                await awsS3Client.send(
+                    new (await import("@aws-sdk/client-s3")).DeleteObjectsCommand({
+                        Bucket: bucketName,
+                        Delete: {
+                            Objects: paths.map(path => {
+                                const { objectName } =
+                                    bucketNameAndObjectNameFromS3Path(path);
+                                return {
+                                    Key: objectName
+                                };
+                            })
```

Comment:
This could throw. If it did we would handle the error as a bulk delete failed du to server issue when in reality it's a wrong assumption on our end.  
Move outside the try/catch block.   

It's important both for not casting too large of a net on our error handling, catching unrelated error and also to be intentional for the reader on what exactly we are catching.